scipy/special/xsf/par_cyl.h

```diff
@@ -2,6 +2,11 @@
 
 #include "specfun/specfun.h"
 
+#if defined(__cplusplus)
+#include <cmath>
+using std::isnan;
+#endif
+
```

Comment:
I think it's best not to touch the `xsf` files. I'm in the process of migrating these into a separate repo, https://github.com/scipy/xsf, which will be included as a submodule of both SciPy and CuPy. I've already migrated all of the commits over, so any change here will be lost.

Just to mention some things though:

Everything in `scipy/special/xsf` is a header in a header only C++ library. We can't have unqualified `using` statements because they'd bleed into anyone's code who uses the `xsf` library.

Also, although not done yet for this particular file, the intention is to have these headers work on both CPU and GPU. We can't have the include of `cmath` here because it will fail in CUDA. Instead, in the `specfun/specfun.h` we have

https://github.com/scipy/scipy/blob/6acf136e6800ca5aab7e3acff9094fc54639e5c1/scipy/special/xsf/specfun/specfun.h#L73

and `config.h` conditionally uses either `<cmath>` or `<cuda/std/cmath>` depending on whether the header is being compiled in a C++ or CUDA project.

I can do an audit of xsf separately and make sure there are no uses of `isnan` and `isinf` outside of the `std` namespace. The way we'll build the test code, these shouldn't even compile.